xbmc/addons/ContextItemAddon.cpp

```diff
@@ -63,6 +60,13 @@ CContextItemAddon::CContextItemAddon(const cp_extension_t *ext)
   }
 }
 
+std::string CContextItemAddon::GetLabel()
+{
+  if (StringUtils::IsNaturalNumber(m_label))
+    return GetString(boost::lexical_cast<int>(m_label.c_str()));
```

Comment:
Only once when the context menu is opened. GetString should already cache it, so I don't think it will be necessary to do that here as well. (that will require an additional flag to keep track of whether it has been translated)